isis/tests/FunctionalTestsChan1m32isis.cpp

```diff
@@ -0,0 +1,558 @@
+  ASSERT_EQ(bandbin["Width"].size(), 256);
+  ASSERT_EQ(bandbin["FilterNumber"].size(), 256);
+  ASSERT_EQ(bandbin["OriginalBand"].size(), 256);
+
+  ASSERT_DOUBLE_EQ(bandbin["Center"][0].toDouble(), 446.02);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][64].toDouble(), 1084.8);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][128].toDouble(), 1723.5899999999999);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][255].toDouble(), 2991.17);
+
+  ASSERT_DOUBLE_EQ(bandbin["Width"][0].toDouble(), 12.31);
+  ASSERT_DOUBLE_EQ(bandbin["Width"][64].toDouble(), 12.29);
+  ASSERT_DOUBLE_EQ(bandbin["Width"][128].toDouble(), 12.61);
+  ASSERT_DOUBLE_EQ(bandbin["Width"][255].toDouble(), 12.18);
+
+
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][0].toDouble(), 5);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][64].toDouble(), 69);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][128].toDouble(), 133);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][255].toDouble(), 260);
+
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][0].toDouble(), 1);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][64].toDouble(), 65);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][128].toDouble(), 129);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][255].toDouble(), 256);
+
+  // Kernels Group
+  PvlGroup &kernel = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  ASSERT_EQ(int(kernel["NaifFrameCode"]), -86520);
+
+  Histogram *hist = cube.histogram();
+
+  ASSERT_DOUBLE_EQ(hist->Average(), 15.58169779027077);
+  ASSERT_DOUBLE_EQ(hist->Sum(), 47368.361282423139);
+  ASSERT_EQ(hist->ValidPixels(), 3040);
+  ASSERT_DOUBLE_EQ(hist->StandardDeviation(), 2.2696592481066249);
+
+  delete hist;
+}
+
+
+TEST(Chan1m32Isis, Chan1m32IsisTestFowardDescending) {
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/chan1m32isisTEMP.cub";
+  QVector<QString> args = {"from=data/chan1m32isis/forwardDescending/M3G20081129T171431_V03_L1B_cropped.LBL",
+                           "loc=data/chan1m32isis/forwardDescending/M3G20081129T171431_V03_LOC_cropped.IMG",
+                           "obs=data/chan1m32isis/forwardDescending/M3G20081129T171431_V03_OBS_cropped.IMG",
+                           "to=" + cubeFileName };
+
+  UserInterface options(APP_XML, args);
+  try {
+    chan1m32isis(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to ingest Chandrayaan image: " << e.toString().toStdString().c_str() << std::endl;
+  }
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+  // Dimensions Group
+  PvlGroup &dimensions = isisLabel->findGroup("Dimensions", Pvl::Traverse);
+  ASSERT_EQ(int(dimensions["Samples"]), 304);
+  ASSERT_EQ(int(dimensions["Lines"]), 5);
+  ASSERT_EQ(int(dimensions["Bands"]), 3);
+
+  // Pixels Group
+  PvlGroup &pixels = isisLabel->findGroup("Pixels", Pvl::Traverse);
+  ASSERT_EQ(pixels["Type"][0].toStdString(), "Real");
+  ASSERT_EQ(pixels["ByteOrder"][0].toStdString(), "Lsb");
+  ASSERT_EQ(double(pixels["Base"]), 0.0);
+  ASSERT_EQ(double(pixels["Multiplier"]), 1.0);
+
```

Comment:
For subsequent tests outside of the "default" it maybe unnecessary to check these groups as it's often the same for images from the same instrument.